Ocelot-Social-Community/Ocelot-Social
# feat: 🍰Implement Registration Slider

## 🍰 Pullrequest


Implement registration with slider.

Link for:

- invite code: `http://localhost:3000/registration?method=invite-code&inviteCode=AAAAAA`
- invite e-mail: `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`

### Issues


- relates #4092

### Test manually

Public registration

- Cypher for query nonce by e-mail: `MATCH (email:EmailAddress {email: "wolle.huss@pjannto.com"}) RETURN email;`

Should be tested, because of partial code connections:

- change e-mail
  - Cypher for query nonce for change: `MATCH (email:UnverifiedEmailAddress {email: "github@pjannto.com"}) RETURN email;`
- change password
- admin invite
  - Cypher for query nonce by e-mail: `MATCH (email:EmailAddress {email: "wolle.huss@pjannto.com"}) RETURN email;`

### Todo


- frontend
  - [x] change nonce length to 5
  - [x] implement new nonce query
  - [x] slider: public registration not possible
  - [x] invent env `INVITE_REGISTRATION`
  - [x] removed about you
  - [x] remove some agreements which the user has to agree to
  - [x] translate `No Public Registration`
  - [x] `webapp/components/EnterNonce/EnterNonce.vue`
    - locale `enter-invite.form.validations` mentions 6 characters …
  - [x] `webapp/components/EnterNonce/EnterNonce.vue`
    - [x] show example code
    - locale `enter-nonce.form.validations` mentions 6 characters …
    - [x] show example code
    - [x] should be 5
  - [x] have toaster for wrong invite code and wrong nonce
  - [x] remove main headline and put it on the first slider
  - [x] delete all old registration possibilities:
    - [x] file: `webapp/pages/registration/create-user-account.vue`
    - [x] file: `webapp/pages/registration/enter-nonce.vue`
    - [x] file: `webapp/pages/registration/signup.vue`
    - [x] file: `webapp/components/Registration/CreateUserAccount.spec.js`
    - [x] file: `webapp/components/Registration/CreateUserAccount.story.js`
    - [x] file: `webapp/components/Registration/CreateUserAccount.vue`
   - [x] more files … ?
   - [x] delete locales as well
     - [x] `termsAndConditions.termsAndConditionsConfirmed`
     - [x] `components.registration.signup.form.data-privacy`
     - [x] `components.registration.signup.form.minimum-age`
     - [x] `components.registration.signup.form.no-commercial`
     - [x] `components.registration.signup.form.no-political`
  - keep files, because they are used:
    - used in password-reset `webapp/pages/password-reset/enter-nonce.vue`:
      - file: `webapp/components/EnterNonce/EnterNonce.vue`
    - used in admin/invite `webapp/pages/admin/invite.vue`:
      - file: `webapp/components/Registration/Signup.spec.js`
      - file: `webapp/components/Registration/Signup.vue`
  - [x] (create issue) write frontend tests (see issue #4311 )
  - [x] (create issue) use make password visible like in LoginForm? (see issue #4312 )
- backend
  - [x] invent env `INVITE_REGISTRATION`
  - [x] implement test for mutation `SignupVerification` with `inviteCode` (see issue #4313 )
  - [ ] fix backend test in `slugifyMiddleware`
    - seems a flaky test?
- e-mails
  - [x] change admin e-mail invite to `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`
  - [x] change public registration e-mail to `http://localhost:3000/registration?method=invite-mail&email=wolle.huss%40pjannto.com&nonce=64853`
  - [x] check e-mail content is the `nonce` visible enough?
- general
  - [x] remove `SignupByInvitation`
 - deployment
  - [x] invent env `INVITE_REGISTRATION`
